public/components/query_compare/search_result/index.tsx

```diff
@@ -104,10 +104,15 @@ export const SearchResult = ({ http }: SearchResultProps) => {
           body: JSON.stringify(requestBody),
         })
         .then((res) => {
-          setQueryResult1(res.result1);
-          updateComparedResult1(res.result1);
-          setQueryResult2(res.result2);
-          updateComparedResult2(res.result2);
+          if (res.result1) {
+            setQueryResult1(res.result1);
+            updateComparedResult1(res.result1);
+          }
+
+          if (res.result2) {
+            setQueryResult2(res.result2);
+            updateComparedResult2(res.result2);
+          }
 
```

Comment:
will there be new tests for these new conditionals?

Reply:
We would like to have snapshot and integration testing for each component, but unfortunately only https://github.com/opensearch-project/dashboards-search-relevance/blob/main/public/components/query_compare/__test__/create_index.test.tsx is available. I can include something in this PR to address this search_result component

Reply:
feeling like if the improvement isn't made now, then when? thank you!

Reply:
Create an issue to track it https://github.com/opensearch-project/dashboards-search-relevance/issues/191